test/unittests/state_bloom_test.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "../utils/utils.hpp"
+#include <evmc/evmc.hpp>
+#include <gtest/gtest.h>
+#include <test/statetest/statetest.hpp>
+
+using namespace evmc::literals;
+using namespace evmone::state;
+
+TEST(state_bloom, combine_blooms)
+{
+    // https://sepolia.etherscan.io/block/2946594
+
+    auto block_bloom =
+        "0x000000000000000000000000800000000000000000000002000000000"
+        "000000000000000000000000000008000000000000000000000000000000040000000000000000000000000000"
+        "000200010000000000000000000004000000000000000000000004000000000000000000000000000000000000"
+        "0000000000000000000000000000c0000200800000000000000000000000000000000000000000000000000800"
+        "000000000000000000000000000000000000000000000000000000040000004000000000000000020020000000"
+        "000000000000000060000000000000000000000020000000000000000000000000000000000000100000000000"
+        "00000"_hex;
+
+    auto bloom_1 =
+        "0x0000000000000000000000008000000000000000000000000000000000000"
+        "000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+        "000000000000000000000000c00002008000000000000000000000000000000000000000000000000008000000"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000"
+        "000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000"
+        "0"_hex;
+
+    const auto bloom_2 =
+        "0x0000000000000000000000008000000000000000000000020000000000000"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+        "000000000000000000000000c00002008000000000000000000000000000000000000000000000000008000000"
+        "000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+        "0"_hex;
+
+    const auto bloom_4 =
+        "0x0000000000000000000000000000000000000000000000000000000000000"
+        "000000000000000000000000080000000000000000000000000000000000000000000000000000000000000000"
+        "000100000000000000000000040000000000000000000000040000000000000000000000000000000000000000"
+        "000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+        "000000000000000000000000000000000000000000000000000000000040000000000000000200000000000000"
+        "000000000000600000000000000000000000000000000000000000000000000000000000001000000000000000"
+        "0"_hex;
+
+    TransactionReceipt r1;
+    memcpy(r1.logs_bloom_filter.bytes, bloom_1.data(), 256);
```

Comment:
You can make a helper, e.g. `bloom_filter_from_bytes()` and use `std::copy` or `std::copy_n`.